docs/proposal-007-apply.md

```diff
@@ -0,0 +1,173 @@
+
+We make it explicit that `apply` is experimental, very open to feedback and
+subject to change.
+
+### Ownership
+
+Ownership comes into play most importantly with deletion.
+
+#### `eksctl`-managed resources
+
+Given the config:
+
+```
+nodeGroups:
+  - name: ng-2
+    desiredCapacity: 1
+```
+
+we discover an existing nodegroup `ng-1` with the `eksctl.io/owned: true` tag (for example),
+do we delete it?
+
+The proposed answer is yes, it should be deleted. With `apply` we assume that all
+previous changes initiated by `eksctl` were initiated through the same config
+file/reconciliation process and that this nodegroup must have been
+deleted from this file with the intention of deleting it from the cluster.
+
+##### Non-authoritative resources
+
+There is precedent for adding the ability to specify resources in a
+non-authoritative list. For example, the [`google` terraform provider supports IAM with 3 different levels of authoritativeness](
+https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/storage_bucket_iam#google_storage_bucket_iam_policy#iam-policy-for-cloud-storage-bucket).
+
+Initial `apply` support will focus on authoritative resource specification.
+
+#### Non-taggable resources
+
+Some resources may not be taggable as owned by `eksctl`.
+If there weren't a `tags` field on addons, for example. `eksctl`-created
+addons would then be indistinguishable from addons created in the AWS console.
+There are at least 2 ways to deal with this case:
+
+##### `eksctl` owns everything
+
+With this option, we assume that `eksctl` owns the set of such resources and
+delete any not listed in the config.
+
+##### `terraform`-style state
+
+Storing some kind of state/config "somewhere" is another option and has been discussed in
+https://github.com/weaveworks/eksctl/issues/642
+
+#### Ignore resource option
+
+An `ignore` flag (cf `terraform`'s `prevent_destroy`) in the schema for all objects
+could be added to tell `eksctl` about such objects and prevent deletions
+and potentially skip reconciliation.
+
+The user would prefix their list:
+
+```
+nodeGroups:
+  - name: ng-1
+    ignore: true
+  - name: ng-2
+    desiredCapacity: 1
+```
+
+#### non-`eksctl`-managed resources
+
+The behavior for resources created outside of `eksctl` could also
+be toggleable by settings in the config/flags on `apply`.
+
+### New config
+
+Proposed is creating a new API version, starting with `v1alpha6`, where, as we add
+`apply` support for cluster/nodegroup properties, we allow for aggressive reevaluation
+of the existing config structure that would hold these newly supported properties/aspects
+of the cluster/nodegroup.
+
+If we decide, _for example_, to add support for reconciling tags, we might
+decide it fits better as a top level key. We would then, with `v1alpha6`, remove
+`tags` under `metadata` and make it a top level field. It should be possible to
+automatically rewrite the previous version into the current version.
```

Comment:
Would the new eksctl apply work with clusters that were created with version v1alpha5?

Reply:
Yes, definitely!